tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs

```diff
@@ -171,85 +179,75 @@ public async Task<StartResult> StartAsync(ValidatorConfiguration configuration)
+
+                var valueChangesPerNodeId = _valueChangeCounterPerNodeId.Stop();
+                var allExpectedValueChanges = true;
+                if (_currentConfiguration.ExpectedValueChangesPerTimestamp > 0) {
+
+                    // TODO collect "expected" parameter as groups related to OPC UA nodes
+                    allExpectedValueChanges = valueChangesPerNodeId?
+                        .All(kvp => (_totalValueChangesCount / kvp.Value ) ==
+                            _currentConfiguration.ExpectedValueChangesPerTimestamp
+                        ) ?? false;
+                    _logger.LogInformation("All expected value changes received: {AllExpectedValueChanges}",
+                        allExpectedValueChanges);
+                }
 
-        /// <summary>
-        /// Stops the Event Hub Client and deregisters event handlers.
-        /// </summary>
-        /// <returns></returns>
-        private async Task StopEventProcessorClientAsync()
-        {
-            _logger.LogInformation("Stopping monitoring of events...");
-
-            if (_client != null)
-            {
-                var tempClient = _client;
-                _client = null;
-                await tempClient.StopProcessingAsync();
-                tempClient.PartitionInitializingAsync -= Client_PartitionInitializingAsync;
-                tempClient.ProcessEventAsync -= Client_ProcessEventAsync;
-                tempClient.ProcessErrorAsync -= Client_ProcessErrorAsync;
+                var incompleteTimestamps = _missingValueChangesChecker.Stop();
+
+                var incrCheckerResult = _incrementalIntValueChecker.Stop();
+
+                var incrSequenceResult = _incrementalSequenceChecker.Stop();
+
+                var stopResult =  new StopResult() {
+                    ValueChangesByNodeId = new ReadOnlyDictionary<string, int>(valueChangesPerNodeId ?? new Dictionary<string, int>()),
+                    AllExpectedValueChanges = allExpectedValueChanges,
+                    TotalValueChangesCount = _totalValueChangesCount,
+                    AllInExpectedInterval = missingTimestampsCounter == 0,
+                    StartTime = _startTime,
+                    EndTime = endTime,
+                    MaxDelayToNow = maxMessageProcessingDelay.ToString(),
+                    MaxDeliveyDuration = maxMessageDeliveryDelay.ToString(),
+                    DroppedValueCount = incrCheckerResult.DroppedValueCount,
+                    DuplicateValueCount = incrCheckerResult.DuplicateValueCount,
+                    DroppedSequenceCount = incrSequenceResult.DroppedValueCount,
+                    DuplicateSequenceCount = incrSequenceResult.DuplicateValueCount,
+                    ResetSequenceCount = incrSequenceResult.ResetsValueCount,
+                    RestartAnnouncementReceived = _restartAnnouncementReceived,
+                };
+
+                return Task.FromResult(stopResult);
+            }
+            finally {
```

Comment:
>finally [](http://example.com/codeflow?start=12&length=7)

sw.Stop()

Reply:
We don't really need to do this as we are not going to restart it.